test/runtests.jl

```diff
@@ -1,88 +1,169 @@
-    @test .1 > @elapsed @test update_julia("1.7.0-rc1", dry_run = true) == v"1.7.0-rc1"
-    @test update_julia("1.7.0-rc1", dry_run = true, verbose = false) == v"1.7.0-rc1"
-    t_before_force_fetch = UpdateJulia.last_fetched[]
-    @test update_julia("1.7.0-rc3", fetch = true) == v"1.7.0-rc3"
-    t_after_force_fetch = UpdateJulia.last_fetched[]
-    @test 0 < t_before_force_fetch < t_after_force_fetch < time()
-    @test update_julia("1.7.0-rc1") == v"1.7.0-rc1"
-    global v_nightly = update_julia("nightly")
-    @test v_nightly >= v"1.8-DEV" && v_nightly > v_latest && v_nightly.prerelease == ("DEV",)
+@testset "versions" begin
+    @test VersionNumber[v"0.1.2", v"0.3.1", v"0.5.0-rc2", v"0.6.0-pre.beta", v"0.7.0-beta",
+        v"1.0.0", v"1.1.1", v"1.2.0-rc2", v"1.2.0-rc3", v"1.2.0", v"1.3.0-rc3", v"1.4.1",
+        v"1.4.2", v"1.5.1", v"1.5.4", v"1.6.0-rc3", v"1.6.1", v"1.6.4", v"1.7.0-beta2",
+        v"1.7.0-beta3"] ⊊ keys(UpdateJulia.versions[])
 end
 
-@testset "Preserve old versions" begin
-    @test ismissing(UpdateJulia.version_of("this-should-not-be-a-valid-command-gj493fD62l4"))
-    @test UpdateJulia.version_of("julia") == UpdateJulia.latest()
-
-    @static if Sys.iswindows()
-        @test_skip "interactive windows installer"
-    else
-        @test UpdateJulia.version_of("julia-1.4") == v"1.4.2"
-        @test UpdateJulia.version_of("julia-1.2") == v"1.2.0"
-        @test UpdateJulia.version_of("julia-1.0.4") == v"1.0.4"
-        @test UpdateJulia.version_of("julia-1.0.0") == v"1.0.0"
-        @test UpdateJulia.version_of("julia-1.0") == v10_latest
+@testset "dry tests" begin
+    @test update_julia(dry_run = true) == UpdateJulia.latest()
+    @suppress_out @test UpdateJulia.update_julia("nightly", dry_run=true, verbose=false) ==
+        UpdateJulia.update_julia(string(UpdateJulia.nightly_version[]), dry_run=true, verbose=false) ==
+        UpdateJulia.nightly_version[]
+end
+
+if ("CI" => "true") ∈ ENV
+
+    @testset "quick test" begin
+        @test update_julia() == UpdateJulia.latest()
+        v_nightly = update_julia("nightly")
+        @test v_nightly >= v"1.8-DEV" && v_nightly > UpdateJulia.latest() && v_nightly.prerelease == ("DEV",)
     end
 
-    @test UpdateJulia.version_of("julia-1.5.1") == v"1.5.1"
-    @test UpdateJulia.version_of("julia-1.5.2") == v"1.5.2"
-    @test UpdateJulia.version_of("julia-1.5.3") == v"1.5.3"
-    @test UpdateJulia.version_of("julia-1.7.0-rc1") == v"1.7.0-rc1"
-    @test UpdateJulia.version_of("julia-1.7.0-rc3") == v"1.7.0-rc3"
-    @test UpdateJulia.version_of("julia-1.5") ∈ [v"1.5.1", v"1.5.3"] # Could be 1.5.1 or 1.5.3 depending on path order b.c. of coexisting systemwide and user installations
-    @test update_julia("1.5", systemwide=false) >= v"1.5.3" # whatever the case, clean up when done
-    @test UpdateJulia.version_of("julia-1.7") >= v"1.7.0-rc3"
-    @test UpdateJulia.version_of("julia") == v_latest
+    #@testset "randomized tests" begin
```

Comment:
What's the deal with this? Do we really want to override Test's random seeding for reproducibility?

Perhaps we seed based on Sys.MACHINE & VERSION to better utilize processing power?

This is a tradeoff between coverage and reproducibility. Let's do both. First curated, then reproducible, then random.